### PR TITLE
Add plugins category with supported integrations

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Adds Plugins section in Sensu Go docs, which contains:
- Install plugins (combines Install plugins and Use assets to install plugins)
- Developer guidelines (includes develop guidelines, getting started, testing, and release process content from Plugins docs)
- Assets reference from Go docs
- Plugins reference from Plugins docs
- Supported integrations category (to highlight features of supported Sensu handler plugins)

## Motivation and Context
Docs IA project

Note: still need to update links (many may be broken in the guides and references) and set up redirects in static.json

### CODE REVIEW

**Comment by hillaryfraley** [comment]
- [x] Remove developer guidelines and refer devs to https://github.com/sensu-plugins/community
Developers should develop according to https://github.com/sensu-community/sensu-plugin-sdk
- [x] Delete https://sensu-docs-plugins-sect-rxplct.herokuapp.com/sensu-go/latest/plugins/plugins/#sensu-plugin-gem
- [x] In plugins reference, answer how to use your old ruby plugins with Sensu Go (sensu ruby runtime)
- [x] Remove https://sensu-docs-plugins-sect-rxplct.herokuapp.com/sensu-go/latest/plugins/plugins/#plugin-configuration
- [x] In https://sensu-docs-plugins-sect-rxplct.herokuapp.com/sensu-go/latest/plugins/plugins/#example, add subheading "Example plugin in Ruby" -- ask someone to draft a basic plugin in Go using the plugin SDK (https://github.com/sensu-community/sensu-plugin-sdk)

**Comment by hillaryfraley** [comment]
List of integrations (check off when DA review is complete):

- [x] PagerDuty
- [x] Slack
- [x] Elasticsearch
- [x] Email
- [x] Jira
- [x] InfluxDB
- [x] Opentsdb
- [x] Wavefront
- [x] timescaledb
~Moogsoft~
- [x] AWS - EC2
- [x] Servicenow
- [x] Saltstack
- [x] Rundeck
- [x] Ansible
- [x] Chef
- [x] Puppet
- [x] Prometheus

**Comment by jspaleta** [comment]
@hillaryfraley Okay I've pushed updates for all outstanding integrations except moogsoft

File: content/sensu-go/6.1/plugins/supported-integrations/ansible.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+{{% notice note %}}
+**NOTE**: The Sensu Ansible Handler plugin is an example of Sensu's auto-remediation integrations.
+To find more integrations, search [Bonsai, the Sensu asset index](https://bonsai.sensu.io/).
+{{% /notice %}}
+
+## Features
+
+The Sensu Ansible Handler plugin supports both Ansible Tower and Ansible AWX implementations of the Ansible Tower API, authenticating using Ansible Tower API tokens.  
+
+- Specify Ansible Tower job templates for remediation actions for checks with Sensu annotations.
```
Comment (hillaryfraley) [inline content/sensu-go/6.1/plugins/supported-integrations/ansible.md]: @jspaleta is this the same thing as the "optional job template requests" described in the fourth bullet point below?

File: content/sensu-go/6.1/plugins/plugins.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+### Go plugin tool and templates
+
+Use the [Sensu Plugin Tool][4] along with a [default template][5] to create the scaffolding for check, handler, mutator, and sensuctl plugins in the Go programming language.
+Follow the instructions in the Sensu Plugin Tool repository to generate a new plugin project based on the template repositories.
+The plugin template repositories wrap the [Sensu Plugin SDK][8], which provides the framework for building Sensu Go plugins.
+
+For a step-by-step walkthrough, read [How to publish an asset with the Sensu Go SDK][7] &mdash; you'll learn how to create a check plugin and a handler plugin with the Sensu Plugin SDK.
+You can also watch our 30-minute webinar, [Intro to assets with the Sensu Go SDK][6], and learn to build a check plugin.
+
```
Comment (hillaryfraley) [inline content/sensu-go/6.1/plugins/plugins.md:116]: Would it be a good idea to add the example I created using the Sensu Plugin Tool here? --->

{{< code go >}}
package main

import (
	"fmt"
	"log"

	"github.com/sensu-community/sensu-plugin-sdk/sensu"
	"github.com/sensu/sensu-go/types"
)

// Config represents the check plugin config.
type Config struct {
	sensu.PluginConfig
	Example string
}

var (
	plugin = Config{
		PluginConfig: sensu.PluginConfig{
			Name:     "check_name",
			Short:    "Description for check_name",
			Keyspace: "sensu.io/plugins/check_name/config",
		},
	}

	options = []*sensu.PluginConfigOption{
		&sensu.PluginConfigOption{
			Path:      "example",
			Env:       "CHECK_EXAMPLE",
			Argument:  "example",
			Shorthand: "e",
			Default:   "",
			Usage:     "An example string configuration option",
			Value:     &plugin.Example,
		},
	}
)

func main() {
	check := sensu.NewGoCheck(&plugin.PluginConfig, options, checkArgs, executeCheck, false)
	check.Execute()
}

func checkArgs(event *types.Event) (int, error) {
	if len(plugin.Example) == 0 {
		return sensu.CheckStateWarning, fmt.Errorf("--example or CHECK_EXAMPLE environment variable is required")
	}
	return sensu.CheckStateOK, nil
}

func executeCheck(event *types.Event) (int, error) {
	log.Println("executing check with --example", plugin.Example)
	return sensu.CheckStateOK, nil
}
{{< /code >}}

...Or is this not the right file to use as an example?

Comment (hillaryfraley) [reply content/sensu-go/6.1/plugins/plugins.md:116]: I am just going to go for it and try to add part of this as an example on the theory that it's usually easier to point out what's wrong.

File: content/sensu-go/6.1/plugins/supported-integrations/ansible.md (hunk 1 above)
Comment (hillaryfraley) [reply content/sensu-go/6.1/plugins/supported-integrations/ansible.md]: Clarification from Jef:
```
related... the first one is basic functionality...     the second bullet is more expansive...  we could rewrite the first one as...
Specify default Ansible Tower job template for remediation actions for all checks, and then use check annotations to override the default on  a check by check basis if needed.
```

File: content/sensu-go/6.1/plugins/supported-integrations/_index.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+Integrations are service-specific and have different setup and configuration requirements.
+Each integration has self-contained documentation with in-depth information about how to install and use it.
+Many of the supported integrations include curated quick-start templates that you only need to edit to match your configuration.
+
+Although this category focuses on our most popular supported integrations, you can find more supported-, Enterprise-, and community-tier integrations at [Bonsai, the Sensu asset index][1].
+
+## Alerting and incident management
+
+- [EC2][2]
```
Comment (cwjohnston) [inline content/sensu-go/6.1/plugins/supported-integrations/_index.md]: Move into deregistration section

File: content/sensu-go/6.1/plugins/supported-integrations/chef.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+[Dynamic runtime assets][5] are shareable, reusable packages that make it easier to deploy Sensu plugins.
+
+## Configuration management
+
+Use the official [Chef Cookbook for Sensu Go][3] for configuration management for your Sensu instance.
+
+
+[1]: ../../../observability-pipeline/observe-process/handlers/
+[2]: ../../../observability-pipeline/observe-process/handler-templates/
+[3]: https://github.com/sensu/sensu-go-chef
```
Comment (cwjohnston) [inline content/sensu-go/6.1/plugins/supported-integrations/chef.md]: Link to cookbook on Chef Supermarket

File: content/sensu-go/6.1/plugins/supported-integrations/puppet.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+[Dynamic runtime assets][5] are shareable, reusable packages that make it easier to deploy Sensu plugins.
+
+## Configuration management
+
+Use the partner-supported [Sensu Puppet module][3] for configuration management for your Sensu instance.
+
+
+[1]: ../../../observability-pipeline/observe-process/handlers/
+[2]: ../../../observability-pipeline/observe-process/handler-templates/
+[3]: https://github.com/sensu/sensu-puppet
```
Comment (cwjohnston) [inline content/sensu-go/6.1/plugins/supported-integrations/puppet.md]: Link to module on Puppet Forge

File: content/sensu-go/6.1/plugins/supported-integrations/ansible.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+The [documentation site][8] includes installation instructions, example playbooks, and module references
+
+
+[1]: ../../../observability-pipeline/observe-process/handlers/
+[2]: ../../../observability-pipeline/observe-process/handler-templates/
+[3]: https://github.com/sensu-community/monitoring-pipelines/blob/latest/remediation/ansible-tower.yaml
+[4]: https://bonsai.sensu.io/assets/sensu/sensu-ansible-handler
+[5]: ../../assets
+[6]: ../../../commercial/
+[7]: https://github.com/sensu/sensu-go-ansible
```
Comment (cwjohnston) [inline content/sensu-go/6.1/plugins/supported-integrations/ansible.md]: Link to collection on Ansible Galaxy

File: content/sensu-go/6.1/plugins/supported-integrations/influxdb.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+You can also use the Sensu InfluxDB Handler integration to create metrics from Sensu status check results for long-term storage in InfluxDB.
+
+{{% notice note %}}
+**NOTE**: The Sensu InfluxDB Handler plugin is an example of Sensu's status and metrics processing and storage integrations.
+To find more integrations, search [Bonsai, the Sensu asset index](https://bonsai.sensu.io/).
+{{% /notice %}}
+
+## Features
+
+- Transform metrics to InfluxDB format: extract and transform the metrics you collect from different sources like Graphite, OpenTSDB, Nagios, and Influx and populate them into InfluxDB.
```
Comment (cwjohnston) [inline content/sensu-go/6.1/plugins/supported-integrations/influxdb.md]:
```suggestion
- Transform metrics to InfluxDB format: extract and transform the metrics you collect from different sources in formats like Graphite, OpenTSDB, Nagios, and Influx and populate them into InfluxDB.
```

Comment (cwjohnston) [reply content/sensu-go/6.1/plugins/supported-integrations/influxdb.md]: This phrasing should be updated in pages for other integrations in "Status and metrics processing and storage"

File: content/sensu-go/6.1/plugins/supported-integrations/wavefront.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+{{% notice note %}}
+**NOTE**: The Sensu Wavefront Handler plugin is an example of Sensu's status and metrics processing and storage integrations.
+To find more integrations, search [Bonsai, the Sensu asset index](https://bonsai.sensu.io/).
+{{% /notice %}}
+
+## Features
+
+- Transform metrics to Wavefront format: extract and transform the metrics you collect from different sources like Graphite, OpenTSDB, Nagios, and Influx and populate them into Wavefront.
+- Specify additional tags to include when processing metrics.
```
Comment (cwjohnston) [inline content/sensu-go/6.1/plugins/supported-integrations/wavefront.md]: This feature of the handler made me think of `output_metric_tags` functionality we recently added to Sensu itself. 

If it is the case that `output_metrics_tags` "just works" with any of our existing handlers, we may want to include a bulletpoint like this one with a link to the relevant documentation on each TSDB integration page.

File: content/sensu-go/6.1/plugins/supported-integrations/_index.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+- [Ansible][8]
+- [Rundeck][9]
+- [SaltStack][10]
+
+## Deregistration
+
+- [Chef][11]
+- [Puppet][12]
+
+## Status and metrics processing and storage
```
Comment (cwjohnston) [inline content/sensu-go/6.1/plugins/supported-integrations/_index.md]:
```suggestion
## Time-series and long-term event storage
```

Consider renaming this section? Time-series or TSDB is a pretty common term which folks might grok more immediately.

**Review by cwjohnston — CHANGES_REQUESTED** [review]
Please find some suggestions based on our mutual review. Other high-level suggestions:

* Rephrase "avoid exposing" usernames and passwords as "keep secure"
* Is there a monitoring-pipelines example for email?
* Move securing with env vars and secrets to the bottom of the features list for most or all of the integrations

**Comment by hillaryfraley** [comment]
@cwjohnston I double-checked https://github.com/sensu-community/monitoring-pipelines but I couldn't find any template for email.

File: content/sensu-go/6.1/plugins/supported-integrations/wavefront.md (hunk 8 above)
Comment (hillaryfraley) [reply content/sensu-go/6.1/plugins/supported-integrations/wavefront.md]: I asked about this in #engineering

File: content/sensu-go/6.1/plugins/supported-integrations/puppet.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+The Sensu Puppet Keepalive Handler plugin is a Sensu [handler][1] that deletes a Sensu entity with a failing keepalive check when the entity's corresponding Puppet node no longer exists or is deregistered.
+
+{{% notice note %}}
+**NOTE**: The Sensu Puppet Keepalive Handler plugin is an example of Sensu's deregistration integrations.
+To find more integrations, search [Bonsai, the Sensu asset index](https://bonsai.sensu.io/).
+{{% /notice %}}
+
+## Features
+
+- Use Sensu annotations to override corresponding Puppet node names with Sensu event names.
```
Comment (cwjohnston) [inline content/sensu-go/6.1/plugins/supported-integrations/puppet.md]:
```suggestion
- Use Sensu annotations to override Sensu entity names with corresponding Puppet node names.
```

File: content/sensu-go/6.1/plugins/supported-integrations/chef.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+{{% notice note %}}
+**NOTE**: The Sensu Chef Handler plugin is an example of Sensu's deregistration integrations.
+To find more integrations, search [Bonsai, the Sensu asset index](https://bonsai.sensu.io/).
+{{% /notice %}}
+
+## Features
+
+- Keep your sensitive API authentication information secure with Sensu [environment variables and secrets management][6].
+
```
Comment (cwjohnston) [inline content/sensu-go/6.1/plugins/supported-integrations/chef.md:23]:
```suggestion
- Use Sensu annotations to override Sensu entity names with corresponding Chef node names.

```

Comment (cwjohnston) [reply content/sensu-go/6.1/plugins/supported-integrations/chef.md:23]: Added in 0.2.0 release now available

**Review by cwjohnston — CHANGES_REQUESTED** [review]

**Review by majormoses — APPROVED** [review]
LGTM

**Pull request merged**